src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -247,6 +248,68 @@ public async Task ReloadTypesAsync(CancellationToken cancellationToken = default
         }
     }
 
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping<T>(string? dataTypeName = null) => TryGetMapping(typeof(T), dataTypeName);
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping(Type? type = null, string? dataTypeName = null)
+    {
+        if (type is null && string.IsNullOrEmpty(dataTypeName))
+            throw new ArgumentException($"Either {nameof(type)} or {nameof(dataTypeName)} must be specified.");
+
+        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+        if (SerializerOptions is null)
+            using (OpenConnection()) { }
+
+        return TryGetMappingCore(type, dataTypeName);
+    }
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public ValueTask<PgTypeInfo?> TryGetMappingAsync<T>(string? dataTypeName = null, CancellationToken cancellationToken = default)
```

Comment:
I think that for cases where we don't actually need a generic type parameter, i.e. a Type parameter is enough (and will likely always be enough), there's no reason to introduce the additional generic API.